src/v/config/configuration.cc

```diff
@@ -3717,6 +3717,56 @@ configuration::configuration()
       "Iceberg is enabled, do not change this value.",
       {.needs_restart = needs_restart::yes, .visibility = visibility::user},
       "redpanda-iceberg-catalog")
+  , iceberg_catalog_type(
+      *this,
+      "iceberg_catalog_type",
+      "Type of an Iceberg catalog that Redpanda will use to commit table "
+      "metadata updates. Supported types: 'rest', 'filesystem'",
+      {.needs_restart = needs_restart::yes, .visibility = visibility::user},
+      datalake_catalog_type::filesystem,
+      {datalake_catalog_type::rest, datalake_catalog_type::filesystem})
+  , iceberg_rest_catalog_endpoint(
+      *this,
+      "iceberg_rest_catalog_endpoint",
+      "URL of Iceberg REST catalog endpoint",
+      {
+        .needs_restart = needs_restart::yes,
+        .example = "http://hostname:8181",
+        .visibility = visibility::user,
+      },
+      std::nullopt)
+  , iceberg_rest_catalog_user_id(
+      *this,
+      "iceberg_rest_catalog_user_id",
+      "Iceberg REST catalog user id. The user id is going to be used to query "
+      "the catalog API for the OAuth token. Required if catalog type is set to "
+      "`rest`",
+      {.needs_restart = needs_restart::yes, .visibility = visibility::user},
+      std::nullopt)
+  , iceberg_rest_catalog_secret(
+      *this,
+      "iceberg_rest_catalog_secret",
+      "Secret to authenticate against Iceberg REST catalog. Required if "
+      "catalog type is set to `rest`",
+      {.needs_restart = needs_restart::yes,
+       .visibility = visibility::user,
+       .secret = is_secret::yes},
+      std::nullopt)
+  , iceberg_rest_catalog_token(
+      *this,
+      "iceberg_rest_catalog_token",
+      "Token that will be used to access the REST Iceberg catalog. If the "
+      "token is present the credentials will be ignored",
+      {.needs_restart = needs_restart::yes,
+       .visibility = visibility::user,
+       .secret = is_secret::yes},
+      std::nullopt)
+  , iceberg_rest_catalog_request_timeout_ms(
+      *this,
+      "iceberg_rest_catalog_request_timeout_ms",
+      "Timeout used when sending requests to REST Iceberg catalog",
```

Comment:
Could we phrase it more like "Maximum length of time that Redpanda waits for a response from the REST catalog before aborting the request"